src/Idris/Parser.idr

```diff
@@ -710,7 +711,7 @@ mutual
                            val <- expr pdef fname indents
                            pure (n, val))
            atEnd indents
-           (n, val) <- pure b.val
+           let (n, val) = b.val
```

Comment:
I'm very excited about this change. I think there are other places where this pattern is used because the type inference used to give up on `let` inside do-notation. Have you found and tried to fix other instances of this pattern?

Reply:
I tried and it didn't work in other places, still because of type inference

Reply:
Hm, interesting. Since when `let ... = b.val` started to work? 
I also once found myself completely lost in the parser code with that pesky problem being the cause of inference errors.

Reply:
It could be that this one always worked but we got accustomed to program
defensively against this type of inference issue and just did not even try.
IME the boolean flag stating whether a parser is productive is always quite
hard to infer.